### PR TITLE
AppBar: Use appBarText for text/icon colors

### PR DESCRIPTION
**Pull request by nucleogenesis**
## Summary

- NavBar & NavbarLink now take & drill a `textColor` prop
- In AppBar, the new token `appBarText` is passed to `textColor`
- In AppBar, it is also used as the default color for icons, the page title, and the user's info

This is done to permit plugins to customize the text in the AppBar so that they can ensure proper contrast with their chosen `appBar` token's value, which is used as the background.

## References

Fixes https://github.com/learningequality/kolibri-instant-schools-plugin/issues/205

#### QA

Are there any regressions in the default Kolibri bevhaior for the top yellow AppBar? Check each plugin and any sub-routes you can.

The fixes that this allows will be tested upon rebuilding the VF IS plugin w/ this patched on Kolibri.

#### Code

- Anything I'm missing where this should be used?
- I have a patch ready for KDSv4 as well that isn't _necessary for this to work_ but it would keep v4 updated w/ a default. I have included fallbacks for each place where `appBarText` is used so having them defined in KDS may not be necessary (cc @AlexVelezLl)
- I have tested this by including the `appBarText` key in the `tokenMapping` section of the theme definition in the Instant Schools plugin and w/ that build, all of the colors are shown as expected. Should this be part of the theme itself - or does adding to that mapping jive w/ the way we want to use the theme machinery?

### CODE REVIEW

File: kolibri/core/assets/src/views/AppBar.vue
Comment (nucleogenesis) [inline kolibri/core/assets/src/views/AppBar.vue]: What the usage looks like in the plugin:

https://github.com/learningequality/kolibri-instant-schools-plugin/pull/209/commits/4f968b5707e8f82ffec818c64d1e4f167d84739c

**Review by pcenov — CHANGES_REQUESTED** [review]
Hi @nucleogenesis - for some reason the top navigation bar is not displayed at all:

https://github.com/user-attachments/assets/5da2eeeb-5c9e-45fc-b131-b6012e2f8b3b

There are multiple `TypeError: Cannot read properties of undefined (reading 'appBar')` errors in the console:

File: kolibri/core/assets/src/views/Navbar/NavbarLink.vue
Comment (rtibbles) [inline kolibri/core/assets/src/views/Navbar/NavbarLink.vue]: Could we avoid prop drilling here, and just reference the value directly? NavbarLink is _only_ used in the Navbar component, and we remove it from the public API completely in develop, so we can just set it directly to the value we want.

File: kolibri/core/assets/src/views/Navbar/index.vue
Comment (rtibbles) [inline kolibri/core/assets/src/views/Navbar/index.vue]: Similarly here, Navbar is _only_ used in AppBar, and is also removed from the public API in develop, so I think we can just reference theme variables directly.

**Review by rtibbles — COMMENTED** [review]
I think we can do some small clean up and avoid any `||` statements by just referencing theme config directly in components.

**Comment by nucleogenesis** [comment]
@pcenov thank you and apologies for the oversight on my end here. I had my devserver running while working on it but didn't see the top bar disappear while making the updates 😅 

I've pushed some updates, so should be ready for re-review.

File: kolibri/core/assets/src/views/Navbar/NavbarLink.vue
Comment (nucleogenesis) [reply kolibri/core/assets/src/views/Navbar/NavbarLink.vue]: Good catch - just updated this

**Review by rtibbles — COMMENTED** [review]
Changes make sense to me, and look like a clean minimal diff. Will wait on manual QA for final approval!

**Review by pcenov — APPROVED** [review]
Thanks @nucleogenesis - no regressions observed while manually testing!

File: kolibri/core/assets/src/styles/themeSpec.js
Comment (rtibbles) [inline kolibri/core/assets/src/styles/themeSpec.js:52]: We don't need to update it here - but if we are going to remove it from the themeTokens in KDS, might just want this to map to something more directly.

File: kolibri/core/assets/src/views/Navbar/NavbarLink.vue
Comment (rtibbles) [inline kolibri/core/assets/src/views/Navbar/NavbarLink.vue:32]: Note to self - this import path will need to be updated when we merge up.

**Review by rtibbles — APPROVED** [review]
No concerns here.

**Pull request merged**
